Feature/rheadley/01.2020.translations

lifewithryan opened the pull request:
# Critical Changes

# Changes
## New Support for Spanish (Mexico)
We've added support for translations in Spanish (Mexico). Now, EDA supports translation in Spanish, Spanish (Mexico), and French, as well as Catalan as a platform-only language.

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Notes

lifewithryan commented on src/objectTranslations/Account-ca.objectTranslation:
```diff
@@ -7,7 +7,7 @@
     </fields>
     <fields>
-        <label>Primary Contact</label>
+        <label>_SYSTEM: One2OneContact</label>
```
note to ryan: we should undo this, it needs to be Primary Contact--my bad.

lifewithryan commented on src/objectTranslations/Contact-ca.objectTranslation:
```diff
@@ -3471,66 +3471,38 @@
         <help>Quin número de telèfon s&apos;hauria d&apos;utilitzar per a la majoria de comunicacions que incloguin aquest contacte?</help>
         <label>Telèfon preferit</label>
         <name>PreferredPhone__c</name>
-        <picklistValues>
```
Ryan i think we are going to have problems here too.

lifewithryan replied:
I actually wanna pull @skosgi1 into this bit. Can you verify if this is what it is supposed to be? I feel like this is going to break translation work that we did.

skosgi1 replied:
@lifewithryan Sorry noticing this just now...looking into it...FYI @robertbwatson please don't merge this yet.

lifewithryan commented on src/objectTranslations/Course_Offering__c-en_GB.objectTranslation:
```diff
@@ -48,7 +48,7 @@
     <fields>
         <label>Time Block</label>
         <name>Time_Block__c</name>
-        <relationshipLabel>Time Blocks</relationshipLabel>
```
Ryan this needs to be the  Time Blocks not Course Offerings.

lifewithryan replied:
I am wrong

lifewithryan commented on src/objectTranslations/Course_Offering__c-es.objectTranslation:
```diff
@@ -48,7 +48,7 @@
     <fields>
         <label>Bloque de tiempo</label>
         <name>Time_Block__c</name>
-        <relationshipLabel>Bloques de tiempo</relationshipLabel>
```
Bloques de Tiempo is the source of truth here.

lifewithryan commented on src/objectTranslations/Course_Offering__c-es_MX.objectTranslation:
```diff
@@ -0,0 +1,62 @@
+    <fields>
+        <label>Bloque de tiempo</label>
+        <name>Time_Block__c</name>
+        <relationshipLabel>Ofertas de cursos</relationshipLabel>
```
Ryan This is wrong here so perhaps change to placeholder and they'll get it next time.

lifewithryan commented on src/objectTranslations/Course_Offering__c-fr.objectTranslation:
```diff
@@ -41,14 +41,14 @@
         <label>Bloc de temps</label>
         <name>Time_Block__c</name>
-        <relationshipLabel>Blocs de temps</relationshipLabel>
+        <relationshipLabel>Choix de modules</relationshipLabel>
```
Blocs de temps should be source of truth here.

lifewithryan replied:
i may be wrong, will verify with team.

lifewithryan replied:
I am wah-rong!

robertbwatson commented:
@lifewithryan @juliesnow Do we need to call out anything in the release notes? It looks like we added support for Mexican Spanish?

lifewithryan commented:
> @lifewithryan @juliesnow Do we need to call out anything in the release notes? It looks like we added support for Mexican Spanish?

good call, that is a worthy call out.

lifewithryan commented:
> > @lifewithryan @juliesnow Do we need to call out anything in the release notes? It looks like we added support for Mexican Spanish?
> 
> good call, that is a worthy call out.

@juliesnow @robertbwatson I've added a very basic callout for the new translation -- but I'm sure that is not how we want to word it.

robertbwatson commented on src/objectTranslations/Behavior_Involvement__c-ca.objectTranslation:
```diff
@@ -9,55 +9,55 @@
     <fields>
-        <help><!-- The Case that represents the event being reported. --></help>
-        <label><!-- Case --></label>
+        <help>El cas que representa l’esdeveniment denunciat.</help>
```
@lifewithryan Just out of curiousity, is the ordering here how they are delivering it to us now? If so, that makes such a WORLD of difference!

lifewithryan replied:
I'd have to double check the original branch. I think they might be doing that now but i still had to do an upload and retrieve to fix other items.

lifewithryan replied:
and the answer is no, this was a result of deploying and retrieving.

skosgi1 commented on src/objectTranslations/Contact-es.objectTranslation:
```diff
@@ -3471,66 +3471,38 @@
-        <picklistValues>
-            <masterLabel>Home</masterLabel>
-            <translation>Teléfono particular</translation>
-        </picklistValues>
```
@lifewithryan @robertbwatson looks like translations for old picklist values are completely gone.

skosgi1 commented on src/objectTranslations/Contact-es.objectTranslation:
```diff
@@ -3471,66 +3471,38 @@
             <masterLabel>Mobile Phone</masterLabel>
-            <translation>Teléfono móvil</translation>
+            <translation><!-- Mobile Phone --></translation>
         </picklistValues>
```
Also, looks like translations for new picklist values have been removed.

robertbwatson commented on unpackaged/post/case_behavior_record_types/package.xml:
```diff
@@ -8,5 +8,13 @@
         	<members>Case.Behavior Management</members>
         	<name>BusinessProcess</name>
 	</types>
+    <types>
```
Any reason to not also do this for `unpackaged/post/facility_display_name`?

lifewithryan replied:
ahhh good catch. I think I removed them from the original pull but forgot to add it to unmanaged. I'll see if we have a translation for it and make sure to put it in.

lifewithryan replied:
I had to add placeholders.

robertbwatson commented on src/package.xml:
```diff
@@ -1097,12 +1126,16 @@
     <types>
         <members>Attribute__c.Credential</members>
         <members>Attribute__c.Student_Characteristic</members>
+        <members>Course_Enrollment__c.Default</members>
```
These got removed when I did a retrieve from the server, so I've addressed this already.

lifewithryan replied:
@robertbwatson github is confusing me, is there any action needed here on my part?

robertbwatson replied:
No, resolved it on my own with the commit I made last night

lifewithryan commented on src/objectTranslations/Course_Offering__c-en_US.objectTranslation:
```diff
@@ -48,7 +48,7 @@
         <label><!-- Time Block --></label>
         <name>Time_Block__c</name>
-        <relationshipLabel><!-- Course Offerings --></relationshipLabel>
+        <relationshipLabel><!-- Time Blocks --></relationshipLabel>
```
my bad, I thought we changed them from Course Offerings to Time Blocks, do I have that info backwards?

lifewithryan replied:
I'm getting myself so confused. Yes they worked on these before we made the fix so they'll have to get those in the next go-round is my guess.

lifewithryan commented on src/objectTranslations/Course_Offering__c-es_MX.objectTranslation:
```diff
@@ -0,0 +1,62 @@
+    <fields>
+        <label>Bloque de tiempo</label>
+        <name>Time_Block__c</name>
+        <relationshipLabel>Bloques de tiempo</relationshipLabel>
```
this is my fault, I thought they changed to time blocks by us but they had old values still reading Course Offerings.

lifewithryan replied:
oh, nevermind, yes on th es-MX file, they wouldn't have had this change so they will have to grab this next time.

lifewithryan commented on src/package.xml:
```diff
@@ -817,21 +817,25 @@
         <members>Affl_Mappings__c-en_US</members>
         <members>Affl_Mappings__c-es</members>
+        <members>Affl_Mappings__c-es_MX</members>
         <members>Affl_Mappings__c-fr</members>
```
> Do we want placeholders for Application, Test, & Test Score?

@robertbwatson I thought those were getting added in that PR?

lifewithryan replied:
could have sworn I just typed an answer here...i thought I saw at least the Test and Test Score translations in your PR for Test and Test score?
In any case, I don't think we'll need them will we? When they do their pull for the next round, they should see they don't currently have translations for them.

robertbwatson replied:
Ah, nevermind - I didn't see them here the first time. They can add the mexican translation files next time, but the others are there now at least.

skosgi1 commented:
@lifewithryan @robertbwatson fixed translations for preferred email and phone. can you take a quick look when you get a moment? Thx.

robertbwatson commented:
@lifewithryan @skosgi1 I think this looks pretty great now, thanks for your hard work on this.

I only have one more remaining question/comment in the src/objectTranslations/Contact-es_MX.objectTranslation file regarding translation of the Preferred Phone/Preferred Email picklist values for a new language. I wasn't sure if we needed to do anything there re: old values...

Otherwise, I think we're okay to merge. I am a little hesitant though, with how much manual work had to be done, and especially with the Preferred Phone/Preferred Email translations...

lifewithryan commented:
> @lifewithryan @skosgi1 I think this looks pretty great now, thanks for your hard work on this.
> 
> I only have one more remaining question/comment in the src/objectTranslations/Contact-es_MX.objectTranslation file regarding translation of the Preferred Phone/Preferred Email picklist values for a new language. I wasn't sure if we needed to do anything there re: old values...
> 
> Otherwise, I think we're okay to merge. I am a little hesitant though, with how much manual work had to be done, and especially with the Preferred Phone/Preferred Email translations...

@robertbwatson I understand that feeling all too well. this is our first round of translation merging since fixing that bug. If it break again, we can adjust. I think right now it might not work for es_MX until we get those translations (obviously). I think it is going to be one of those things that just has to lag behind a bit. Now that it's fixed and updated though, I would hope it is not an issue moving forward.

lifewithryan commented:
@robertbwatson it still says "changes requested" so I want to be sure I've addressed everything. I am ready for you to merge upon your approval. (No pressure or anything...) :)
